### PR TITLE
init

### PR DESCRIPTION
**Pull request by MaksymKos**
https://maksymkos.github.io/react_clock/

### CODE REVIEW

File: src/Clock.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+type Props = {
+  clockName: string,
+  hasClock: boolean,
+};
+
+export class Clock extends React.Component<Props> {
+  state = {
+    time: new Date(),
+    timerId: 0,
+  };
```
Comment (Viktor-Kost) [inline src/Clock.tsx:13]: The timerId should not be in the state because it does not affect rendering.

File: src/App.tsx [hunk 2]
```diff
@@ -1,39 +1,66 @@
...
+
+  render() {
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+
+        {this.state.hasClock && (
+          <Clock
+            clockName={this.state.clockName}
+            hasClock={this.state.hasClock}
```
Comment (Viktor-Kost) [inline src/App.tsx:60]: The 'hasClock' prop is unnecessary in the 'Clock' component as the 'Clock' will only be rendered when 'hasClock' is true.

File: src/App.tsx [hunk 3]
```diff
@@ -1,39 +1,66 @@
...
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends React.Component {
+  state = {
+    clockName: 'Clock-0',
+    hasClock: true,
+    timerId: 0,
```
Comment (Viktor-Kost) [inline src/App.tsx:16]: The timerId should not be in the state because it does not affect rendering.

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 4]
```diff
@@ -1,39 +1,66 @@
...
+export class App extends React.Component {
+  state = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentDidMount(): void {
+    const timerIdInternal = window.setInterval(() => {
```
Comment (IvanFesenko) [inline src/App.tsx:21]: No need in extra variable
```suggestion
    timerId = window.setInterval(() => {
```

File: src/App.tsx [hunk 5]
```diff
@@ -1,39 +1,66 @@
...
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+      this.setState({ clockName: name });
+    }, 3300);
 
-        {' time is '}
+    timerId = timerIdInternal;
```
Comment (IvanFesenko) [inline src/App.tsx:31]:
```suggestion

```

File: src/Clock.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    time: new Date(),
+  };
+
+  componentDidMount(): void {
+    const timerIdInternal = window.setInterval(() => {
+      this.setState({ time: new Date() });
+      console.info(this.state.time.toUTCString().slice(-12, -4));
+    }, 1000);
+
+    timerId = timerIdInternal;
```
Comment (IvanFesenko) [inline src/Clock.tsx:21]: Same here

**Review by IvanFesenko — APPROVED** [review]

**Pull request closed without merging**